### PR TITLE
Add support for Domoticz MQTT integration

### PR DESCRIPTION
**Pull request by gneluka**
Adding basic support for Domoticz MQTT integration.
Changes:
- Added field for configuring "Domoticz In" topic under MQTT section of config page
- Added "Parameter per number sequence" select and "Domoticz Counter Idx" field under MQTT section of config page
- Updated javascript functions as needed to support these new fields: read, write and update parameters.
- Updated ClassFlowMQTT to publish to the configured "Domoticz In" topic using the correct payload format for Domoticz counters.
- Minor changes and updates to interface_mqtt and server_mqtt to support the added functionality.

### CODE REVIEW

File: param-docs/parameter-pages/MQTT/DomoticzTopicIn.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+# Parameter `DomoticzTopicIn`
+Default Value: `domoticz/in`
+
+MQTT IN topic for Domoticz integration, under which the counters are published.
+
+Counter Idx is required.
```
Comment (caco3) [inline param-docs/parameter-pages/MQTT/DomoticzTopicIn.md:6]: does this refer to `Parameter <NUMBER>.DomoticzIDX`? If so, please name it that ways

File: param-docs/parameter-pages/MQTT/NUMBER.DomoticzIDX.md [hunk 2]
```diff
@@ -0,0 +1,4 @@
+# Parameter `<NUMBER>.DomoticzIDX`
+Default Value: ``
```
Comment (caco3) [inline param-docs/parameter-pages/MQTT/NUMBER.DomoticzIDX.md:2]: Or should it be zero?

File: param-docs/parameter-pages/MQTT/DomoticzTopicIn.md (hunk 1 above)
Comment (gneluka) [reply param-docs/parameter-pages/MQTT/DomoticzTopicIn.md]: It does. I have updated the parameter name as requested. Thanks

File: param-docs/parameter-pages/MQTT/NUMBER.DomoticzIDX.md (hunk 2 above)
Comment (gneluka) [reply param-docs/parameter-pages/MQTT/NUMBER.DomoticzIDX.md]: Actually there is no default value for the Idx in Domoticz. Either ''(null string) or zero are invalid Idx values in Domoticz, A valid Idx for a Domoticz device, is an integer greater than zero. I have changed it to zero just to indicate that a numeric value is expected. 
Thanks

**Comment by caco3** [comment]
I finally had a look on it.
The UI misses the default values:
grafik
(I am not sure how to add them, need to find out first).

**Comment by caco3** [comment]
Thanks for your PR!
I added some minor changes. I did not really test it as I do not have Domoticz.
I will merge it. Please test and check it all works as needed.

**Comment by gneluka** [comment]
> Thanks for your PR! I added some minor changes. I did not really test it as I do not have Domoticz. I will merge it. Please test and check it all works as needed.

Thanks so much. And thank you for the amazing work you have done on this project.

**Pull request merged**